### PR TITLE
Add docstrings across the whole project

### LINKED ISSUE #13: Integrate new hook in place of PEP257 in pre-commit's config
See: https://github.com/chewse/pre-commit-mirrors-pydocstyle

### PR DESCRIPTION
**Pull request by makkaronis4e**
Fix manage.py and pre-commit config
Fix #13

### CODE REVIEW

**Review by webknjaz — CHANGES_REQUESTED** [review]
Mixed changes

**Comment by webknjaz** [comment]
> Fix config for tests

It's a lie: you need to make code PEP257-compliant

**Comment by webknjaz** [comment]
@makkaronis4e 

here's a few docs for you to read:
- [x] https://www.python.org/dev/peps/pep-0257/
- [x] http://blog.dolphm.com/pep257-good-python-docstrings-by-example/

Also, exclude migrations folder from pep257 check.

**Comment by webknjaz** [comment]
> `feature/fix-config`

You should've used `feature/13-fix-config` as per convention you agreed to follow in #8

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -21,8 +21,10 @@
   sha: v2.1.1
   hooks:
   - id: pydocstyle
+  exclude: blog/migrations/
```
Comment (webknjaz) [inline .pre-commit-config.yaml]: for this to work you should indent it by 2 spaces

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -21,8 +21,10 @@
...
 
 - repo: git://github.com/Lucas-C/pre-commit-hooks
   sha: v1.1.4
   hooks:
   - id: remove-tabs
+  exclude: blog/migrations/
```
Comment (webknjaz) [inline .pre-commit-config.yaml]: same as above

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -17,12 +17,16 @@
...
-#  sha: v0.3.2
-#  hooks:
-#  - id: pep257
+- repo: git://github.com/chewse/pre-commit-mirrors-pydocstyle
+  sha: v2.1.1
+  hooks:
+  - id: pydocstyle
+    exclude: blog/migrations/
+
+
```
Comment (webknjaz) [inline .pre-commit-config.yaml]: plz remove one line here.

**Review by webknjaz — CHANGES_REQUESTED** [review]
And get rid of bogus commits.

File: manage.py [hunk 4]
```diff
@@ -1,4 +1,5 @@
 #!/usr/bin/env python
+"""Script for managing django."""
```
Comment (webknjaz) [inline manage.py]: ..app

File: mysite/__init__.py [hunk 5]
```diff
@@ -0,0 +1 @@
+"""Configure files for site."""
```
Comment (webknjaz) [inline mysite/__init__.py]: Why is it in actionable mood?

File: mysite/urls.py [hunk 6]
```diff
@@ -1,18 +1,4 @@
...
-    1. Add an import:  from my_app import views
-    2. Add a URL to urlpatterns:  url(r'^$', views.home, name='home')
-Class-based views
-    1. Add an import:  from other_app.views import Home
-    2. Add a URL to urlpatterns:  url(r'^$', Home.as_view(), name='home')
-Including another URLconf
-    1. Import the include() function: from django.conf.urls import url, include
-    2. Add a URL to urlpatterns:  url(r'^blog/', include('blog.urls'))
-"""
+"""mysite URL Configuration."""
```
Comment (webknjaz) [inline mysite/urls.py]: what is mysite here?

File: blog/models.py [hunk 7]
```diff
@@ -1,8 +1,11 @@
+"""Models configuration in blog."""
 from django.db import models
 from django.utils import timezone
 
 
 class Post(models.Model):
+    """Configuration of the post class."""
```
Comment (webknjaz) [inline blog/models.py]: It's not configuration, and it's not for class. It's a model definition.

File: blog/apps.py [hunk 8]
```diff
@@ -1,5 +1,8 @@
+"""Configuration module for blog."""
```
Comment (webknjaz) [inline blog/apps.py]: No, it's an apps config.

File: blog/models.py [hunk 9]
```diff
@@ -1,8 +1,11 @@
+"""Models configuration in blog."""
```
Comment (webknjaz) [inline blog/models.py]: It's not a config, but a description for models for interfacing with the DB.

File: blog/admin.py [hunk 10]
```diff
@@ -1,3 +1,4 @@
+"""Admin panel registration."""
```
Comment (webknjaz) [inline blog/admin.py]: No, it's configuration for admin panel.

File: blog/__init__.py [hunk 11]
```diff
@@ -0,0 +1 @@
+"""Configure files for blog."""
```
Comment (webknjaz) [inline blog/__init__.py]: It should not be in imperative mood: just importing this package won't emit any actions. This is a description, no a command.

File: blog/models.py [hunk 12]
```diff
@@ -12,8 +15,10 @@ class Post(models.Model):
             blank=True, null=True)
 
     def publish(self):
+        """Publish post in blog."""
         self.published_date = timezone.now()
         self.save()
 
     def __str__(self):
+        """Return title of the post."""
```
Comment (webknjaz) [inline blog/models.py]: @makkaronis4e in Python, "dunder methods" (wrapped w/ double underscores, also known as "magic methods") are special methods responsible for certain behaviours of objects.
`obj.__str__()` is called if the object is being stringified, for example using `str(obj)`, meaning rendering the object into a string representation.

That said, the docstring would be smth like `Render the post instance from its title.`

**Review by webknjaz — CHANGES_REQUESTED** [review]
- [ ] Improve PR name
- [ ] Make commit messages reflect their contents

File: blog/tests.py [hunk 13]
```diff
@@ -1,3 +1,4 @@
+"""Test for ptoject."""
```
Comment (webknjaz) [inline blog/tests.py]: Typo: `ptoject`

Comment (webknjaz) [inline blog/tests.py]: Test suite for the project.

File: mysite/__init__.py [hunk 14]
```diff
@@ -0,0 +1 @@
+"""Configuration files for site."""
```
Comment (webknjaz) [inline mysite/__init__.py]: Not files: python operates such things as modules and packages. Also config files are normally some structured text files, which *don't contain any code*.

File: blog/models.py [hunk 15]
```diff
@@ -1,8 +1,11 @@
+"""Description for models for interfacing the DB."""
 from django.db import models
 from django.utils import timezone
 
 
 class Post(models.Model):
+    """Definition of the model."""
```
Comment (webknjaz) [inline blog/models.py]: I see that from the line above as it inherits model. Don't refer to implementation details, unless they really matter + don't repeat smth obvious one can notice w/o a docstring.
Better tell me that it's a blog post structure declaration.

File: blog/__init__.py [hunk 16]
```diff
@@ -0,0 +1 @@
+"""Configuration files for blog."""
```
Comment (webknjaz) [inline blog/__init__.py]: It's not files.

**Review by webknjaz — CHANGES_REQUESTED** [review]
It looks better now, but there's still a few nitpicks, I'd like you to fix :)

**Comment by makkaronis4e** [comment]
@webknjaz Done

File: blog/views.py [hunk 17]
```diff
@@ -1,3 +1,4 @@
+"""Views configuration for blog."""
```
Comment (webknjaz) [inline blog/views.py]: It's not config. It's just views.

File: blog/__init__.py [hunk 18]
```diff
@@ -0,0 +1 @@
+"""Configuration for blog."""
```
Comment (webknjaz) [inline blog/__init__.py]: It is not a config.

File: mysite/__init__.py [hunk 19]
```diff
@@ -0,0 +1 @@
+"""Configuration for site."""
```
Comment (webknjaz) [inline mysite/__init__.py]: It's not a config. It is an app.

**Review by webknjaz — CHANGES_REQUESTED** [review]

File: blog/__init__.py (hunk 18 above)
Comment (webknjaz) [reply blog/__init__.py]: `Django blog app`

File: blog/views.py (hunk 17 above)
Comment (webknjaz) [reply blog/views.py]: `Views for the blog app.`

File: mysite/__init__.py (hunk 19 above)
Comment (webknjaz) [reply mysite/__init__.py]: `Mysite Django app.`

File: manage.py [hunk 20]
```diff
@@ -11,7 +11,7 @@
         # issue is really that Django is missing to avoid masking other
         # exceptions on Python 2.
         try:
-            import django  # noqa: F401
+        import django # noqa: F401
```
Comment (webknjaz) [inline manage.py]: @makkaronis4e fix this.

**Review by webknjaz — APPROVED** [review]

**Pull request merged**